processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -188,6 +188,22 @@ public BeanMappingMethod build() {
                             Message.BEANMAPPING_NOT_ASSIGNABLE, resultType, method.getResultType()
                         );
                     }
+                    else if ( resultType.isAbstract() ) {
+                        ctx.getMessager().printMessage(
+                            method.getExecutable(),
+                            beanMappingPrism.mirror,
+                            Message.BEANMAPPING_ABSTRACT,
+                            resultType,
+                            method.getResultType()
+                        );
+                    }
+                }
+                else if ( !method.isUpdateMethod() && method.getReturnType().isAbstract() ) {
```

Comment:
You are right. For #657, we said if the result type of the BeanMappingMethod is not an interface then we can consider that the method has a factory method. See this.

I think that with this PR all the errors that you mentioned are covered, just not in one place 😄. Should we structure that a bit better?